### ISSUE TITLE
Packing is not handled correctly in netcdf4 format module

### ISSUE THREAD

**Issue by aph42** [pygeode/pygeode#134]
Reading in NetCDF files in which variables have been packed (using the scale_factor and offset attributes) in the netcdf4 module doesn't work correctly at the moment, because the re-scaling gets applied both by the underlying module as well as by pygeode.

As a minimal example:

    import pygeode as pyg
    import numpy as np

    N = 100
    lin = np.linspace(300, 301, N)
  
    # Create a variable and test reading from it
    t = pyg.yearlessn(N)
    v = pyg.Var((t,), name = 'var', values = lin)

    # Write to netcdf
    pyg.save('packedvar1.nc', v, format='netcdf', pack=True, compress=True)

    # read from netcdf and access data
    d1 = pyg.open('packedvar1.nc', format='netcdf4')

    # This should give back the same values as were written, but the assertion fails
    assert np.allclose(d1.var[:], lin)

**Comment by aph42** [comment]
This has an easy fix - one can turn off the rescaling done by the underlying module. I think that's the right way since the rescaling is done independently of the format module at the moment in pygeode.

In writing the tests to sort this out I found another issue with saving files using netcdf4; individual variables lead to an error. This also is an easy fix of putting `in_dataset` through `asdataset`. I'll commit changes and regression tests shortly.

**Comment by aph42** [comment]
In fact the same issue was happening on write out as well. Resolved with commit 3374046

**Comment by neishm** [comment]
This update seems to break `issue114_test`, more specifically there's an issue with the `asdataset` wrapper added to `netcdf4.save`.  The problem is that `in_dataset` could be a dictionary of group names pointing to a different datasets.  I think the type of `in_dataset` would need to be checked, and if it's a dict it would need to do a loop as is done further below in the code.

**Comment by aph42** [comment]
Ok, I have a fix that calls `asdataset` only after checking for the case that `in_dataset` is a dictionary. In the latter case the values in the dictionary are also passed through `asdataset`, so you could specify groups using, e.g. lists of variables. This does now pass the regression tests. Does this behaviour seem reasonable?

**Comment by neishm** [comment]
Looks good to me!
